### PR TITLE
[rocksdb] Add tbb support

### PR DESCRIPTION
**Pull request by pgoodman**
This lets users use the CLOCK algorithm for their block cache [1], which requires Intel TBB to be linked in.

[1] https://github.com/facebook/rocksdb/wiki/Block-Cache#clock-cache

### CODE REVIEW

File: ports/rocksdb/portfile.cmake [hunk 1]
```diff
@@ -3,8 +3,6 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
   OUT_SOURCE_PATH SOURCE_PATH
   REPO facebook/rocksdb
-  REF v5.17.2
```
Comment (Codiferous) [inline ports/rocksdb/portfile.cmake]: I believe this change will require the user to install `rocksdb` with the `--head` flag:

```
Building package rocksdb[core,tbb,zlib]:x86-windows...
CMake Error at scripts/cmake/vcpkg_from_github.cmake:103 (message):
  Package does not specify REF.  It must built using --head.
```

Is this intentional? What's the reason behind this change?

Comment (pgoodman) [reply ports/rocksdb/portfile.cmake]: This part of the change shouldn't have been included, my bad. I was trying to see if a SIGSEGV was related to the version of RocksDB, or related to my (mis)use of it. It was the latter.

**Comment by Codiferous** [comment]
Okay, I think this is looking good enough to merge now. Thanks for this PR! 😄
